packages/medusa/src/loaders/plugins.ts

```diff
@@ -118,47 +142,91 @@ async function runLoaders(pluginDetails: PluginDetails, container: MedusaContain
   )
 }
 
-function registerMedusaApi(pluginDetails: PluginDetails, container: MedusaContainer): void {
+async function registerMedusaApi(
+  pluginDetails: PluginDetails,
+  container: MedusaContainer
+): Promise<void> {
   registerMedusaMiddleware(pluginDetails, container)
   registerStrategies(pluginDetails, container)
 }
 
-function registerStrategies(pluginDetails: PluginDetails, container: MedusaContainer): void {
-  let module
-  try {
-    const path = `${pluginDetails.resolve}/strategies/tax-calculation`
-    if (existsSync(path)) {
-      module = require(path).default
+function registerStrategies(
+  pluginDetails: PluginDetails,
+  container: MedusaContainer
+): void {
+  const files = glob.sync(`${pluginDetails.resolve}/strategies/[!__]*`, {})
+  const registeredServices = {}
```

Comment:
This doesn't work since the default strategies will be registered prior to this, thus it already has the registration and we wish to override it but only once.